core-derive/tests/expand/simple_struct.expanded.rs

```diff
@@ -0,0 +1,155 @@
+    use _smithy4rs::schema::StaticSchemaShape as _StaticSchemaShape;
+    #[automatically_derived]
+    impl _StaticSchemaShape for SimpleStruct {
+        fn schema() -> &'static _SchemaRef {
+            &SIMPLE_SCHEMA
+        }
+    }
+};
+const _: () = {
+    extern crate smithy4rs_core as _smithy4rs;
+    use _smithy4rs::schema::SchemaRef as _SchemaRef;
+    use _smithy4rs::serde::serializers::Serializer as _Serializer;
+    use _smithy4rs::serde::serializers::SerializeWithSchema as _SerializeWithSchema;
+    use _smithy4rs::serde::serializers::StructSerializer as _StructSerializer;
+    #[automatically_derived]
+    impl _SerializeWithSchema for SimpleStruct {
+        fn serialize_with_schema<S: _Serializer>(
+            &self,
+            schema: &_SchemaRef,
+            serializer: S,
+        ) -> Result<S::Ok, S::Error> {
+            let mut ser = serializer.write_struct(schema, 2usize)?;
+            ser.serialize_member_named("field_a", &FIELD_A, &self.field_a)?;
+            ser.serialize_member_named("field_b", &FIELD_B, &self.field_b)?;
+            ser.end(schema)
+        }
+    }
+};
+#[automatically_derived]
+pub struct SimpleStructBuilder {
+    field_a: Option<String>,
+    field_b: Option<i32>,
+}
+#[automatically_derived]
+impl SimpleStructBuilder {
+    pub fn new() -> Self {
+        Self {
+            field_a: None,
+            field_b: None,
+        }
+    }
+    pub fn field_a(mut self, value: String) -> Self {
+        self.field_a = Some(value);
+        self
+    }
+    pub fn field_b(mut self, value: i32) -> Self {
+        self.field_b = Some(value);
+        self
+    }
+    pub fn build(self) -> Result<SimpleStruct, String> {
+        Ok(SimpleStruct {
+            field_a: self.field_a.ok_or_else(|| "field_a is required".to_string())?,
+            field_b: self.field_b.ok_or_else(|| "field_b is required".to_string())?,
+        })
+    }
+}
+const _: () = {
+    extern crate smithy4rs_core as _smithy4rs;
+    use _smithy4rs::schema::SchemaRef as _SchemaRef;
+    use _smithy4rs::schema::StaticSchemaShape as _StaticSchemaShape;
+    use _smithy4rs::serde::deserializers::Deserializer as _Deserializer;
+    use _smithy4rs::serde::deserializers::DeserializeWithSchema as _DeserializeWithSchema;
+    use _smithy4rs::serde::deserializers::Error as _Error;
+    #[automatically_derived]
+    impl<'de> _DeserializeWithSchema<'de> for SimpleStruct {
+        fn deserialize_with_schema<D>(
+            schema: &_SchemaRef,
+            deserializer: &mut D,
+        ) -> Result<Self, D::Error>
+        where
+            D: _Deserializer<'de>,
+        {
+            let builder = SimpleStructBuilder::new();
+            let builder = deserializer
+                .read_struct(
+                    schema,
+                    builder,
+                    |builder, member_schema, de| {
+                        if std::sync::Arc::ptr_eq(member_schema, &FIELD_A) {
```

Comment:
I dont think we should use ptr_eq here as it would prevent you from using a different, manually created schema definition (for example a manually instantiated document schema)

Reply:
Tabled for later, but we should probably use member index here.